Modules/hashlib.h

```diff
@@ -48,18 +54,30 @@
 
 #include "pythread.h"
 #define ENTER_HASHLIB(obj) \
-    if ((obj)->lock) { \
-        if (!PyThread_acquire_lock((obj)->lock, 0)) { \
-            Py_BEGIN_ALLOW_THREADS \
-            PyThread_acquire_lock((obj)->lock, 1); \
-            Py_END_ALLOW_THREADS \
-        } \
+    if ((obj)->use_mutex) { \
+        Py_BEGIN_ALLOW_THREADS \
```

Comment:
Remove `Py_BEGIN_ALLOW_THREADS` and `Py_END_ALLOW_THREADS` from this macro.

It's not necessary here because `PyMutex_Lock` internally will release the GIL (or equivalent in nogil builds) if it blocks on acquiring the mutex.